Implemented

u11107 opened the pull request:
-Switch theme
-Search history

Commit:
Implemented
-Switch theme
-Search history

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -0,0 +1,39 @@
+
+class App : Application() {
+
+    companion object {
```
Согласно конвенции котлина, companion object [должен находиться в конце класса](https://kotlinlang.org/docs/coding-conventions.html#class-layout)

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -0,0 +1,39 @@
+    companion object {
+        const val EXAMPLE_PREFERENCES = "example_preferences"
+        const val KEY_THEME = "key"
+        lateinit var sharedMemory: SharedPreferences
```
Лучше вынести его в класс, так как companion object статический

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -82,17 +91,48 @@ class SearchActivity : AppCompatActivity() {
+
+    @SuppressLint("NotifyDataSetChanged")
+    private fun focusVisibility(hasFocus:Boolean){
+        binding.tittleHistory.visibility = if (hasFocus && binding.searchEditText
```
можно if вынести в общий, так как здесь и ниже они одинаковые, в теле ифа присваивать видимость обоим элементам

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/SearchHistory.kt:
```diff
@@ -0,0 +1,47 @@
+import com.google.gson.reflect.TypeToken
+
+class SearchHistory {
+    companion object {
```
если цель сделать полный класс статическим, то надо просто вместо class написать object, тогда не потребуется companion

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/SearchHistory.kt:
```diff
@@ -0,0 +1,47 @@
+        }
+
+        fun addTrack(track: Track) {
+            var historyTrackList = fillInList()
```
может быть val

Anna-Malinina commented on app/src/main/java/com/example/playlistmaker/SearchHistory.kt:
```diff
@@ -0,0 +1,47 @@
+        fun addTrack(track: Track) {
+            var historyTrackList = fillInList()
+
+            if (historyTrackList.contains(track)) {
```
можно не проверять на наличие, метод remove вернет true, если смог удалить, а false, если не смог. То есть он сам в себе проверяет

Commit:
Corrected
-Font in the "Clear history" button
-Changed the line "You searched" to "Search history"
-Display search history after clicking on delete button.
